src/devices/Mbi5027/Mbi5027.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.Spi;
+
+namespace Iot.Device.Multiplexing
+{
+    /// <summary>
+    /// MBI5027 16-Bit shift register With 3-State output register and error detection.
+    /// Supports SPI and GPIO control.
+    /// </summary>
+    public class Mbi5027 : ShiftRegister
+    {
+        // Datasheet: http://archive.fairchip.com/pdf/MACROBLOCK/MBI5027.pdf
+        private Mbi5027PinMapping _pinMapping;
+
+        /// <summary>
+        /// Initialize a new shift register connected through GPIO.
+        /// </summary>
+        /// <param name="pinMapping">The pin mapping to use by the binding.</param>
+        /// <param name="bitLength">Bit length of register, including chained registers. Default is 16 bits.</param>
+        /// <param name="gpioController">The GPIO Controller used for interrupt handling.</param>
+        /// <param name="shouldDispose">Option (true by default) to dispose the GPIO controller when disposing this instance.</param>
+        public Mbi5027(Mbi5027PinMapping pinMapping, int bitLength = 16, GpioController gpioController = null,  bool shouldDispose = true)
+        : base(new ShiftRegisterPinMapping(pinMapping.Sdi, pinMapping.Clk, pinMapping.LE, pinMapping.OE), bitLength, gpioController, shouldDispose)
```

Comment:
you could likely still do that with SoftwareSpi, you'd just need to manually pass in GpioController